Failing test: TestInterceptLogger/includes_the_caller_location

Steps to reproduce:

```
% export GOPATH=/tmp/gp
% go get github.com/hashicorp/go-hclog
% cd /tmp/gp/src/github.com/hashicorp/go-hclog
% go test -v ./...
[…]
=== RUN   TestInterceptLogger/includes_the_caller_location
    TestInterceptLogger/includes_the_caller_location: interceptlogger_test.go:287: 
        	Error Trace:	interceptlogger_test.go:287
        	Error:      	Not equal: 
        	            	expected: "[INFO]  go-hclog/interceptlogger.go:76: test: this is test: who=programmer why=\"testing is fun\"\n"
        	            	actual  : "[INFO]  go-hclog/interceptlogger_test.go:280: test: this is test: who=programmer why=\"testing is fun\"\n"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,2 @@
        	            	-[INFO]  go-hclog/interceptlogger.go:76: test: this is test: who=programmer why="testing is fun"
        	            	+[INFO]  go-hclog/interceptlogger_test.go:280: test: this is test: who=programmer why="testing is fun"
        	            	 
        	Test:       	TestInterceptLogger/includes_the_caller_location
    TestInterceptLogger/includes_the_caller_location: interceptlogger_test.go:292: 
        	Error Trace:	interceptlogger_test.go:292
        	Error:      	Not equal: 
        	            	expected: "[INFO]  go-hclog/interceptlogger.go:84: test: this is test: who=programmer why=\"testing is fun\"\n"
        	            	actual  : "[INFO]  go-hclog/interceptlogger_test.go:280: test: this is test: who=programmer why=\"testing is fun\"\n"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,2 @@
        	            	-[INFO]  go-hclog/interceptlogger.go:84: test: this is test: who=programmer why="testing is fun"
        	            	+[INFO]  go-hclog/interceptlogger_test.go:280: test: this is test: who=programmer why="testing is fun"
        	            	 
        	Test:       	TestInterceptLogger/includes_the_caller_location
=== RUN   TestInterceptLogger/supports_resetting_the_output
=== RUN   TestInterceptLogger/supports_resetting_the_output_with_flushing
--- FAIL: TestInterceptLogger (0.00s)
    --- PASS: TestInterceptLogger/sends_output_to_registered_sinks (0.00s)
    --- PASS: TestInterceptLogger/sink_includes_with_arguments (0.00s)
    --- PASS: TestInterceptLogger/sink_includes_name (0.00s)
    --- PASS: TestInterceptLogger/intercepting_logger_can_create_logger_with_reset_name (0.00s)
    --- PASS: TestInterceptLogger/Intercepting_logger_sink_can_deregister_itself (0.00s)
    --- PASS: TestInterceptLogger/Sinks_accept_different_log_formats (0.00s)
    --- PASS: TestInterceptLogger/handles_parent_with_arguments_and_log_level_args (0.00s)
    --- PASS: TestInterceptLogger/derived_standard_loggers_send_output_to_sinks (0.00s)
    --- FAIL: TestInterceptLogger/includes_the_caller_location (0.00s)
    --- PASS: TestInterceptLogger/supports_resetting_the_output (0.00s)
    --- PASS: TestInterceptLogger/supports_resetting_the_output_with_flushing (0.00s)
[…]
```

PR #68 might be related

In general, why not just use `strings.HasSuffix()` instead of a regexp? It’ll be quicker to run and easier to understand.

cc @evanphx @drewbailey

--- a/interceptlogger_test.go
+++ b/interceptlogger_test.go
@@ -284,12 +284,12 @@ func TestInterceptLogger(t *testing.T) {
 		rest := str[dataIdx+1:]
 
 		// This test will break if you move this around, it's line dependent, just fyi
-		assert.Equal(t, "[INFO]  go-hclog/interceptlogger.go:76: test: this is test: who=programmer why=\"testing is fun\"\n", rest)
+		assert.Equal(t, "[INFO]  go-hclog/interceptlogger_test.go:280: test: this is test: who=programmer why=\"testing is fun\"\n", rest)
 
 		str = sbuf.String()
 		dataIdx = strings.IndexByte(str, ' ')
 		rest = str[dataIdx+1:]
-		assert.Equal(t, "[INFO]  go-hclog/interceptlogger.go:84: test: this is test: who=programmer why=\"testing is fun\"\n", rest)
+		assert.Equal(t, "[INFO]  go-hclog/interceptlogger_test.go:280: test: this is test: who=programmer why=\"testing is fun\"\n", rest)
 	})
 
 	t.Run("supports resetting the output", func(t *testing.T) {

--- a/intlogger.go
+++ b/intlogger.go
@@ -10,7 +10,6 @@ import (
 	"log"
 	"os"
 	"reflect"
-	"regexp"
 	"runtime"
 	"sort"
 	"strconv"
@@ -178,8 +177,6 @@ func trimCallerPath(path string) string {
 	return path[idx+1:]
 }
 
-var logImplFile = regexp.MustCompile(`.+intlogger.go|.+interceptlogger.go$`)
-
 // Non-JSON logging format function
 func (l *intLogger) logPlain(t time.Time, name string, level Level, msg string, args ...interface{}) {
 	if len(l.timeFormat) > 0 {
@@ -199,8 +196,7 @@ func (l *intLogger) logPlain(t time.Time, name string, level Level, msg string, 
 		// Check if the caller is inside our package and inside
 		// a logger implementation file
 		if _, file, _, ok := runtime.Caller(3); ok {
-			match := logImplFile.MatchString(file)
-			if match {
+			if strings.HasSuffix(file, "intlogger.go") || strings.HasSuffix(file, "interceptlogger.go") {
 				offset = 4
 			}
 		}

